sdk/core/azure-core/inc/context.hpp

```diff
@@ -39,6 +39,13 @@ namespace Azure { namespace Core {
     };
 
   public:
+
+// The compiler gives a false positive on the union type, disable that warning for this constructor
```

Comment:
I wouldn't call this a false positive: it is in fact the case that no member of the anonymous union is initialized in the default constructor on line 49. I'd say something like `// it is intended that the default constructor initializes no member of the anonymous union`.

Reply:
That's a good point.  Perhaps a better way to address this is to set the union value to 0.

Reply:
> That's a good point. Perhaps a better way to address this is to set the union value to 0.

I went out of my way to devise a way to avoid zeroing an extra byte of memory in `std::optional`'s default constructor, so I may not be the person to ask, but I'm 100% fine with not initializing this anonymous union member. The uninitialized state is in no way unexpected, and it's handled correctly in all of the other member functions.